pkg/util/ingress/ingress.go

```diff
@@ -32,7 +32,7 @@ import (
 
 const (
 	alphaNumericChars = `\-\.\_\~a-zA-Z0-9/`
-	regexEnabledChars = `\^\$\[\]\(\)\{\}\*\+`
+	regexEnabledChars = `\^\$\[\]\(\)\{\}\*\+\|`
```

Comment:
Should `?` also be included here?

Reply:
I think |, ? and \ should all be included.

Reply:
Here is a pcre meta characters reference:

https://www.php.net/manual/en/regexp.reference.meta.php

Reply:
I also think the valid characters for alphaNumeric needs to be updated for non-regex compliance with the RCF spec...
```
	alphaNumericChars = `\-\.\_\~\!\$\&\'\(\)\*\+\,\;\=\:\@\%a-zA-Z0-9/`
	regexEnabledChars = `\^\$\[\]\(\)\{\}\*\+\\\?\|`
```

Reply:
Maybe it's better  just to try to compile regex to validate it?  sure that some non‑standard shortcuts should be covered , but as I can see in current approach, even named capture groups wouldn't work. 

I can do some refactor here if you agree

Reply:
Go uses different regex engine than Nginx so compiling will not really validate it. Nginx is PCRE and Go is RE2.

Reply:
Hey folks, thanks for the PR and sorry for the mess! 

Can you please look at https://github.com/kubernetes/ingress-nginx/pull/9511 as we are turning the characters configurable, but there is a default set?

@boxboatmatt Originally we were going to do following RFC, but this subset is too big for a path, which in case of ingress is not even validated per request query (like "& chars", etc.

That's why our original decision was to reduce this scope at maximum. Anyway, we may be missing some regex chars, (pipe and ? were two) so please take a look into that PR :)

Thanks!